test_agent/rust/src/main.rs

```diff
@@ -44,19 +47,43 @@ fn connect_to_socket(addr: &str, port: u16) -> Result<TcpStream, Box<dyn std::er
     }
 }
 
-async fn connect_and_receive() -> Result<(), Box<dyn std::error::Error>> {
+async fn connect_and_receive(transport_name: &String) -> Result<(), Box<dyn std::error::Error>> {
     let test_agent = connect_to_socket(TEST_MANAGER_ADDR.0, TEST_MANAGER_ADDR.1)?;
     let ta_to_tm_socket = connect_to_socket(TEST_MANAGER_ADDR.0, TEST_MANAGER_ADDR.1)?;
     let foo_listener_socket_to_tm = connect_to_socket(TEST_MANAGER_ADDR.0, TEST_MANAGER_ADDR.1)?;
 
-    let u_transport = UTransportSocket::new()?;
-    debug!("Socket transport created successfully");
+    let u_transport: Box<dyn UTransport> = if "zenoh" == transport_name {
+        let uauthority = UAuthority {
+            name: Some("MyAuthName".to_string()),
+            number: Some(Number::Id(vec![1, 2, 3, 4])),
+            ..Default::default()
+        };
+        let uentity = UEntity {
+            name: "default.entity".to_string(),
+            id: Some(u32::from(rand::random::<u16>())),
+            version_major: Some(1),
+            version_minor: None,
+            ..Default::default()
+        };
+        dbg!("zenoh transport created successfully");
+
+        Box::new(
+            UPClientZenoh::new(Config::default(), uauthority, uentity)
+                .await
+                .unwrap(),
+        )
+    } else {
+        dbg!("Socket transport created successfully");
+        Box::new(UTransportSocket::new()?)
+    };
+
+    let u_transport_ref: &dyn UTransport = &*u_transport;
```

Comment:
If you follow the suggestion [here](https://github.com/eclipse-uprotocol/up-tck/pull/44/files#r1605436552) then you would leave this within the `Box` or `Arc`.

Reply:
same as above, with box